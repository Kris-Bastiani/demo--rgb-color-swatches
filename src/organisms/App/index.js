import React, { useEffect, useState } from 'react';

import { API_BASE_URL } from '../../constants';
import formatColors from '../../helpers/formatColors';
import getData from '../../helpers/getData';
import Button from '../../atoms/Button';
import Swatch from '../../atoms/Swatch';

const updateColors = async (setColors, setSelected) => {
	const response = await getData(`${API_BASE_URL}/list-extended`);
	setColors(formatColors(response));
	setSelected('none');
};

const App = () => {
	const [colors, setColors] = useState([]);
	const [selected, setSelected] = useState('none');

	useEffect(() => { updateColors(setColors, setSelected); }, []);

	return (
		<div className='App'>
			<Button label='Refresh' onClick={() => updateColors(setColors, setSelected)} />
			<p>Selected: {selected}</p>
			{colors.map((color, index) => {
				const { red, green, blue } = color;
				return <Swatch key={index} red={red} green={green} blue={blue} setSelected={setSelected} />;
			})}
		</div>
	);
};

export default App;
