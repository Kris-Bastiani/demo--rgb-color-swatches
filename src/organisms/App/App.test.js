import React from 'react';
import { render } from '@testing-library/react';

import App from '.';

describe('Component: App', () => {
	it('should match snapshot', () => {
		const { container } = render(<App />);
		expect(container).toMatchSnapshot();
	});
});
