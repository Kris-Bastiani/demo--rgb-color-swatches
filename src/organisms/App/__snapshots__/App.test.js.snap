// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App should match snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <button>
      Refresh
    </button>
    <p>
      Selected: 
      none
    </p>
  </div>
</div>
`;
